hashdist/cli/remote_cli.py

```diff
@@ -0,0 +1,304 @@
+                ctx.logger.critical('Refer to documentation and class AppInfoFileRepository to setup pcs_api for a quick test')
+                exit(1)
+            apps_repo = AppInfoFileRepository(app_info_path)
+            if not os.path.exists(user_credentials_path):
+                ctx.logger.critical('Not found any users credentials repository file: ' + `user_credentials_path`)
+                ctx.logger.critical('Refer to documentation and class UserCredentialsFileRepository to setup pcs_api for a quick test')
+                exit(1)
+            user_credentials_repo = UserCredentialsFileRepository(user_credentials_path)
+            provider_name = apps_repo._app_info.keys()[0].split(".")[0]
+            app_info = apps_repo.get(provider_name)
+            user_info = user_credentials_repo.get(app_info)
+            storage = StorageFacade.for_provider(provider_name) \
+                       .app_info_repository(apps_repo,app_info.app_name) \
+                       .user_credentials_repository(user_credentials_repo,user_info.user_id) \
+                       .build()
+            ctx.logger.info("Cloud storage user_id = " + `storage.get_user_id()`)
+            ctx.logger.info("Cloud storage quota = "+`storage.get_quota()`)
+            ctx.logger.info("Cloud storage is  ready")
+            ctx.logger.info("Getting remote manifest")
+
+        if args.objects in ['build','build_and_source']:
+            store = BuildStore.create_from_config(ctx.get_config(), ctx.logger)
+            os.chdir(store.artifact_root)
+
+            #try loading the local copy of the remote manifest
+            try:
+                with open(pjoin("..","build_manifest.json"),"r") as manifest_file:
+                    local_manifest = json.loads(manifest_file.read())
+            except:
+                ctx.logger.warn("Using an empty local manifest because build_manifest.json could not be read")
+                local_manifest={}
+            if args.dryrun:
+                ctx.logger.info("Comparing build store to last local copy of remote manifest")
+                skipping=''
+                pushing=''
+                for package in os.listdir(store.artifact_root):
+                    for artifact in os.listdir(pjoin(store.artifact_root,package)):
+                        if local_manifest.has_key(package) and local_manifest[package].has_key(artifact):
+                            skipping += package+"/"+artifact+" Skipping\n"
+                        else:
+                            pushing += package+"/"+artifact+" Pushing\n"
+                sys.stdout.write(skipping+"(Use --force to push skipped artifacts anyway)\n==============================================\n"+pushing)
+            else:
+                try:
+                    remote_manifest_string = MemoryByteSink()
+                    fpath = CPath('/bld/')
+                    bpath=fpath.add("build_manifest.json")
```

Comment:
Is this the synchronization method?  A file is maintained on the remote describing the contents of its directories?